apps/docs/pages/docs/api-reference/fields/richtext.mdx

```diff
@@ -0,0 +1,733 @@
+| [`controls`](#controls)                   | `controls: { MyControl: {render: (editor, editorState) => <div>My Control</div>} }` | `Record<string, {render: (editor, editorState) => ReactNode}>` | -        |
+| [`textSelectOptions`](#textselectoptions) | `textSelectOptions: ["p","h2"]`                                                     | `("p" \| "h2" \| "h3" \| "h4" \| "h5" \| "h6")[]`              | -        |
+| [`extensions`](#extensions)               | `extensions: [StarterKit, Link]`                                                    | `Extensions[]` <i>from TipTap</i>                              | -        |
+| [`selector`](#selector)                   | `selector: (ctx) => ({ ... })`                                                      | `RichTextSelector`                                             | -        |
+
+## Required params
+
+### `type`
+
+The type of the field. Must be `"richtext"` for RichText fields.
+
+```tsx {6} copy
+const config = {
+  components: {
+    Example: {
+      fields: {
+        description: {
+          type: "richtext",
+        },
+      },
+      // ...
+    },
+  },
+};
+```
+
+## Optional params
+
+### `contentEditable`
+
+Enable inline text editing for this field.
+
+```tsx {7} copy
+const config = {
+  components: {
+    Example: {
+      fields: {
+        description: {
+          type: "richtext",
+          contentEditable: false,
+        },
+      },
+      // ...
+    },
+  },
+};
+```
+
+### `menu`
+
+Provide an override for the grouped button layout rendered in the editor toolbar. Each key in your `menu` matches a menu group and maps to an array of control keys.
+
+```tsx {7-10} copy
+const config = {
+  components: {
+    Example: {
+      fields: {
+        description: {
+          type: "richtext",
+          menu: {
+            typography: ["Bold", "Italic", "Underline"],
+            blocks: ["BulletList", "OrderedList"],
+          },
+        },
+      },
+    },
+  },
+};
+```
+
+### `inlineMenu`
+
+Provide an override for the grouped button layout rendered in the `BubbbleMenu` when inline editing of your field. Each key in your `inlineMenu` matches a menu group and maps to an array of control keys.
+
+```tsx {7-10} copy
+const config = {
+  components: {
+    Example: {
+      fields: {
+        description: {
+          type: "richtext",
+          inlineMenu: {
+            typography: ["Bold", "Italic", "Underline"],
+            blocks: ["BulletList", "OrderedList"],
+          },
+        },
+      },
+    },
+  },
+};
+```
+
+### `controls`
+
+Controls let you swap or extend the buttons available to each menu key. A control is an object with a `render(editor, editorState)` function, so you can plug in any React UI while still receiving the merged TipTap state produced by our selector utilities.
```

Comment:
The control object needs breaking down into another level of docs to cover the render function and its args.

Additionally, I suggest the following for this:

```suggestion
Controls are actions shown in the [`menu`](#menu) and [`inlineMenu`](#inlinemenu). You can define your own controls, or override the [default ones](#default-controls).
```